src/routes/PlayQuiz/index.tsx

```diff
@@ -1,71 +1,111 @@
   const [isPlaying, setIsPlaying] = useState(false);
+  const [isGameStarted, setIsGameStarted] = useState(false);
   const [winner, setWinner] = useState('');
-  const selectedOptionsData = gameInfo.teams.reduce(
-    (acc, team) => acc.concat(team.selectedOptions),
+  const selectedOptionsData: any = gameInfo.teams.reduce(
+    (acc, team) => acc.concat(team.selectedOptions || []),
     [] as SelectedOptions[],
   );
   const attemptedQuestionIds = selectedOptionsData.map((x) => x.questionId);
   const isQuestionAttempted = !!selectedQuestion && attemptedQuestionIds.includes(selectedQuestion.questionId);
   const showQuestionTimer = !!timeLimit && !!selectedQuestion && !isQuestionAttempted;
   const { showModal, getGameData, updateGame } = useStore();
   const [isLoading, setIsLoading] = useState(true);
-  const navigate = useNavigate();
 
   useEffect(() => {
     if (gameId) {
       getGameData(parseInt(gameId)).then((game) => {
+        debugger;
         const { quiz, ...restData } = game;
         const quizData = quiz[0];
-        const categoryIds = quizData.categories.map((category) => category.categoryId);
-        setQuizInfo({
-          quizId: quizData.quizId,
-          name: quizData.name,
-          categoryIds,
-          numberOfQuestionsPerCategory: quizData.numberOfQuestionsPerCategory,
+        let questionNum = 1;
+        let allQuestionsCount = 0;
+
+        quizData.categories.forEach((category) => {
+          category.questions.forEach((q) => {
+            q.questionNum = questionNum;
+            questionNum += 1;
+            allQuestionsCount++;
+          });
         });
 
-        if (categoryIds.length > 0) {
-          setCategoriesInfo(formatCategoryInfo(quizData.categories, categoryIds));
+        setQuizInfo(quizData);
+        setGameInfo(restData);
+        const attemptedQuestionsCount = restData.teams.reduce((count, team) => {
+          count += team.selectedOptions.length;
+          return count;
+        }, 0);
+        const isCompleted = allQuestionsCount === attemptedQuestionsCount;
+
+        if (isCompleted) {
+          const winners = getWinners(restData.teams);
+          const winnerIds = winners.map((winner) => winner.teamId).join(',');
+          setWinner(winnerIds);
+        } else if (attemptedQuestionsCount > 0) {
+          setSelectedQuestion(getLastAttemptedQuestion(restData.teams, quizData.categories));
         }
 
-        setGameInfo(restData);
+        setIsGameStarted(attemptedQuestionsCount > 0);
         setIsLoading(false);
       });
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [gameId]);
 
+  function startGame() {
+    setIsPlaying(!!gameInfo.selectionTimeLimit);
+    setIsGameStarted(true);
+  }
+
+  function getLastAttemptedQuestion(teams: Team[], categories) {
+    const clonedTeams = JSON.parse(JSON.stringify(teams));
+    const teamsWithMaxAttemptedQuestion = clonedTeams.reduce((acc, data) => {
```

Comment:
Are we doing a sort here? can we use array.sort instead of reduce?

Reply:
teamsOrderedByNumOfAttemtedQuestions

Reply:
no, I'm not trying to sort here. I'm trying to auto-select last Attempted question when user refreshes the page in the middle of a game

Reply:
But the reduce here is doing a sort & then selecting the max value using 0 index right?

Reply:
yeah.. removed `reduce` using `Math.max` instead